src/test/java/com/google/sps/BooksTest.java

```diff
@@ -0,0 +1,54 @@
+package com.google.sps.servlets;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.net.URL;
+import org.mockito.Mockito;
+import java.io.IOException;
+import java.util.Map;
+import java.util.HashMap;
+
+/** */
+@RunWith(JUnit4.class)
+public final class BooksTest {
+  
+  public String EMPTY_MAP_STRING = "{}";
+  
+ /**
+  * Checks that if the keyword was nonsense, no result is returned.
+  */
+  @Test
+  public void noOutputTest() throws IOException {
+    Map<String, String[]> badParams = new HashMap<>();
+    String[] gibberish = {"kjdaflkdsalklkjfdsalkflkds"};
+    badParams.put("key", gibberish);
+    SetUpObject booksGet = TestingUtil.mockGet(badParams);
+    BooksServlet booksServlet = new BooksServlet();
```

Comment:
Does this mean that you're calling the real API during testing?

Reply:
Yes. Because I made the tests in an older branch, I didn't want to test specific methods in the servlet, but for future tests it would be ideal to also test the individual methods.